components/broker/engine/engine.go

```diff
@@ -480,9 +473,12 @@ func (brk *broker) findTarget(ctx *BrokerEventContext) (err error) {
 				if ctx.TargetAdapter, err = brk.store.Adapter(ctx, ctx.Event.Target.Name, typ); err != nil {
 					return err
 				}
-				if err := ctx.TargetAdapter.Resolve(ctx.Data); err != nil {
+
+				spec, err := ctx.TargetAdapter.Resolve(ctx.Data)
+				if err != nil {
```

Comment:
I thought we figured out a way for the Broker to rewrite the target which would negate the need for the modifications to the SubscriptionMgr?

Reply:
Updated in https://github.com/xigxog/kubefox/pull/29/commits/cc9526c3a0e1fcfa278d5001efaa9dee2857fa72